go/interpreter/lfvm/instructions_test.go

```diff
@@ -1191,6 +1191,87 @@ func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
 	}
 }
 
+func TestGenericCreate_ReportsErors(t *testing.T) {
+
+	u64Overflow := new(uint256.Int).Add(uint256.NewInt(math.MaxUint64), uint256.NewInt(math.MaxUint64))
+	one := uint256.NewInt(1)
+
+	tests := map[string]struct {
+		offset, size  uint256.Int
+		kind          tosca.CallKind
+		revision      tosca.Revision
+		expectedError error
+	}{
+		"offset overflow": {
+			offset:        *u64Overflow,
+			size:          *one,
+			expectedError: errOverflow,
+		},
+		"size overflow": {
+			offset:        *one,
+			size:          *u64Overflow,
+			expectedError: errOverflow,
+		},
+		"not enough gas for code size": {
+			offset:        *one,
+			size:          *uint256.NewInt(31),
+			revision:      tosca.R12_Shanghai,
+			expectedError: errOutOfGas,
+		},
+		"not enough gas for create 2 init code hashing": {
+			offset:        *one,
+			size:          *one,
+			kind:          tosca.Create2,
+			revision:      tosca.R11_Paris,
+			expectedError: errOutOfGas,
+		},
```

Comment:
These tests look like they are testing specific properties that got introduced by Shanghai and Paris. It should thus be tested that before those releases some worked and after those not -- or the other way around, as needed.

Reply:
max init code size has its own test checking that it does not fail before shanghai. I could add it as another case here with expected error nil and unify both tests. As for the create 2 case, it does not depend on revision, this was just a workaround to not have to add a the gas as a test case parameter as well. I'll think it over